torchtune/_cli/download.py

```diff
@@ -86,11 +99,28 @@ def _add_arguments(self) -> None:
     def _download_cmd(self, args: argparse.Namespace) -> None:
         """Downloads a model from the Hugging Face Hub."""
         # Download the tokenizer and PyTorch model files
+
+        # Raise if local_dir_use_symlinks is invalid
+        output_dir_use_symlinks: Union[Literal["auto"], bool]
+        use_symlinks_lowercase = args.output_dir_use_symlinks.lower()
```

Comment:
`argparse` can actually handle all of this logic within the `add_argument` method.

e.g.

```
self._parser.add_argument(
    ...,
    choices = ["true", "false", "auto"],
)
```

Reply:
This wont work because coz.. `true` and `false` should be of type Boolean.

Reply:
But here the they convert to Boolean, I just used there logic  https://github.com/huggingface/huggingface_hub/blob/54e4f76594d2d8338ddccab566a9750cfe4da1bd/src/huggingface_hub/commands/download.py#L137

Reply:
Ahh I missed that in the code, thanks for pointing it out. It's so gross, but seems like the best way to go forward. 

Thanks for all your work on this @apthagowda97, I'll go ahead and merge!